### ISSUE TITLE
"reference to 'Vector3' is ambiguous" on pip install

### ISSUE THREAD

**Issue by brucedjones** [nmwsharp/robust-laplacians-py#5]
I'm running MacOS 12.0.1 on apples new M1 chip and am not able to install robust-laplacian via pip. It looks like there is no arm wheel, so pip is building from source. The build gets 98% of the way through and errors on "reference to 'Vector3' is ambiguous". 

I'm going to clone and try and debug this locally, but any guidance would be appreciated!

**Comment by brucedjones** [comment]
Cloning the repo and building locally was successful, digging further

**Comment by brucedjones** [comment]
Ok, so it looks like the problem was with the version of Eigen. I installed Eigen using brew which installed version 3.4.0, where robust-laplacian expects version 3.3.5. I manually built and installed 3.3.5 locally and the pip install worked afterwards.

So that unblocked me, but it would be nice to have an arm64 wheel prebuilt. I just forked and am testing doing just that.

**Comment by nmwsharp** [comment]
Thank you for digging in to this!

I should probably track down & fix the reason for the failure on 3.4.0, but in the meantime I agree we should get apple silicon builds out :) I already set this up for another project, so I have the appropriate cibuildwheel commands handy. I just created a PR that adds it #6, assume the build passes they should be available soon.

**Comment by brucedjones** [comment]
oh nice, thanks for jumping on this!

**Comment by nmwsharp** [comment]
no problem! might be a while though, apparently I've forgotten how my CI setup works :)

**Comment by nmwsharp** [comment]
Also, I pushed ce15ef3, which should fix the compile error on Eigen 3.4.0. The new version of Eigen introduced a `Vector3` typedef, which conflicted with some of our own types due to a lazy `using namespace Eigen` which I have since removed.

**Comment by nmwsharp** [comment]
Looks like upgrading pybind to latest fixed those build problems, I'll push a new version

**Comment by nmwsharp** [comment]
And it's live https://pypi.org/project/robust-laplacian/0.2.3/
